File: app/src/main/java/io/madcamp/treasurehunterar/treasure/CollectionService.java
package io.madcamp.treasurehunterar.treasure;

import java.util.List;

import io.madcamp.treasurehunterar.collection.Collection;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface CollectionService {
    @GET("/collection/{collectionNum}")
    Call<Collection> getCollectionById(@Path("collectionNum") String collectionNum);

    @GET("/collections")
    Call<List<Collection>> getCollections();

    @POST("/collection/found/{collectionNum}")
    Call<Object> collectionFound(@Path("collectionNum") String collectionNum);
}

File: app/src/main/java/io/madcamp/treasurehunterar/collection/CollectionService.kt
package io.madcamp.treasurehunterar.collection

import com.jakewharton.retrofit2.converter.kotlinx.serialization.asConverterFactory
import kotlinx.serialization.json.Json
import okhttp3.MediaType.Companion.toMediaType
import retrofit2.Retrofit
import retrofit2.http.GET

interface CollectionService {

    @GET("/collections")
    suspend fun getCollections(): List<Collection>


}

private const val BASE_URL = "http://172.10.5.178"
private val retrofit = Retrofit.Builder()
    .addConverterFactory(Json.asConverterFactory("application/json".toMediaType()))
    .baseUrl(BASE_URL)
    .build()


object CollectionApi {
    val retrofitService : CollectionService by lazy {
        retrofit.create(CollectionService::class.java)
    }
}
